### DISCUSSION TITLE
All tables have been updated in match, is there a bug?

### DISCUSSION THREAD

**Discussion by ru88s** [supabase/supabase#3900] [Questions] [answered]
Normally this is not a problem, but the other day I had a phenomenon where all values were updated.
Is it just me?

```js

    const { data,error } = await supabase.from("comments").insert([{
          talk_id: talk.id,
          name: name,
          body: body,
          res_id: nextResId,
        ]);

        if (data) {
          await updateTalks(nextResId, name, body);
        }

        const updateTalks = async(nextResId: string, name: string, body: string) => {

          const { data, error } = await supabase.from("talks").update({ commentsnum: nextResId, updated_at: new Date()}).match({ id: talk.id });

          if (data) {
            return data;
          } else {
            updateTalks(nextResId, name, body);
          }
        };

```

<img width="568" alt="スクリーンショット 2021-11-15 17 06 51" src="https://user-images.githubusercontent.com/42270511/141744802-d5d69422-7027-49f5-84b4-7cd18068f734.png">

**Comment by awalias** [comment, marked as answer]
@ru88s is this snippet running inside of a loop? If so is there a chance that `nextResId` isn't being incremented correctly in the loop?

**Reply by ru88s** [reply]
@awalias No, there is no loop.
It runs only once after a comment has been successfully posted.
However, I have included a retry process for updateTalks in case of errors.

**Reply by steve-chavez** [reply]
@ru88s Are you sure that the `updateTalks` call is not recursing? You've defined it as a recursive function.

Try isolating the problem. Do a single `supabase.from("talks").update().eq('id', your_id)` and see if it indeed updates all rows of the table without considering the filter.

**Reply by ru88s** [reply]
```js
  const updateTalks = async (nextResId: string, name: string, body: string) => {
      const { data, error } = await supabase
        .from("talks")
        .update({ commentsnum: nextResId, updated_at: new Date() })
        .match({ id: talk.id })

      if (data) {
        return data;
      } else {
        errorMail(error, name, body);
      }
  };
```

The above code caused the bug again.
I decided to change it to `.eq("id", talk.id);` and see what happens.

![スクリーンショット 2022-02-28 21 53 11](https://user-images.githubusercontent.com/42270511/155986648-c1518ed7-4efa-4354-8ac7-30377bed4274.png)

**Reply by steve-chavez** [reply]
Sorry to hear that @ru88s. Do you expect to only have one row affected each time the update is called?

If so, you could prevent this error by adding the [single()](https://supabase.com/docs/reference/javascript/single) modifier, if more than one row would be updated then the modifier will cause the statement to fail and the transaction to rollback(so no data will be modified).

```js
      const { data, error } = await supabase
        .from("talks")
        .update({ commentsnum: nextResId, updated_at: new Date() })
        .match({ id: talk.id })
        .single()
```

**Reply by ru88s** [reply]
@steve-chavez 
Oh, I see.
I'll try using `single()`.
Thank you very much for your help.
